### PR TITLE
add sample filter

### PR DESCRIPTION
**Pull request by allieychen**
In BQ to VCF pipeline, add one option for the user to load the selected samples.

Issues: [85](https://github.com/googlegenomics/gcp-variant-transforms/issues/85)
Tested: unit tests & manually ran BQ to VCF.

### CODE REVIEW

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 1]
```diff
@@ -496,3 +496,9 @@ def add_arguments(self, parser):
               'chromosome is requested. Only variants matching at least one '
               'of these regions will be loaded. If this parameter is not '
               'specified, all variants will be kept.'))
+    parser.add_argument(
+        '--sample_names',
```
Comment (arostamianfar) [inline gcp_variant_transforms/options/variant_transform_options.py]: nit: s/sample_names/call_names to be consistent with naming.

File: gcp_variant_transforms/bq_to_vcf.py [hunk 2]
```diff
@@ -169,9 +170,14 @@ def _bigquery_to_vcf_shards(
     variants = (p
                 | 'ReadFromBigQuery ' >> beam.io.Read(bq_source)
                 | bigquery_to_variant.BigQueryToVariant(annotation_names))
-    call_names = (variants
-                  | 'CombineCallNames' >>
-                  combine_call_names.CallNamesCombiner())
+    if known_args.sample_names:
```
Comment (arostamianfar) [inline gcp_variant_transforms/bq_to_vcf.py]: Please add a TODO to modify the SQL query to only return the specified call names to reduce the amount of data being processed from the beginning (I think the query would be more complex, so it's fine to leave as is for now).

Comment (allieychen) [reply gcp_variant_transforms/bq_to_vcf.py]: That's a good point! Add a TODO for now.

File: gcp_variant_transforms/bq_to_vcf.py [hunk 3]
```diff
@@ -158,7 +159,7 @@ def _bigquery_to_vcf_shards(
   Also, it writes the meta info and data header with the call names to
   `vcf_header_file_path`.
   """
-
+  # TODO (allieychen): Modify the SQL query with the specified call_names.
```
Comment (arostamianfar) [inline gcp_variant_transforms/bq_to_vcf.py]: nit: remove space before `(`

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 4]
```diff
@@ -496,3 +496,9 @@ def add_arguments(self, parser):
               'chromosome is requested. Only variants matching at least one '
               'of these regions will be loaded. If this parameter is not '
               'specified, all variants will be kept.'))
+    parser.add_argument(
+        '--call_names',
+        default=None, nargs='+',
+        help=('A list of call names (separated by a space). Only variant '
```
Comment (arostamianfar) [inline gcp_variant_transforms/options/variant_transform_options.py]: nit: `Only variants for these calls will be ...` and also, s/samples/calls at the end.

File: gcp_variant_transforms/libs/bigquery_vcf_schema_converter.py [hunk 5]
```diff
@@ -264,9 +264,13 @@ def _add_info_fields_from_alternate_bases(schema, infos):
   Notice that the validation of field mode is skipped for reserved fields since
   the mode (NULLABLE) of field in alternate bases is expected to be different
   from the mode (REPEATED) in reserved field definition.
+
+  Any `Record` field within alternate bases is considered as an annotation
+  field, and the annotation fields are skipped.
   """
   for field in schema.fields:
-    if field.name in _CONSTANT_ALTERNATE_BASES_FIELDS:
+    if (field.name in _CONSTANT_ALTERNATE_BASES_FIELDS or
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/bigquery_vcf_schema_converter.py]: This seems unrelated to this PR; usually it's better to do it in another PR, but I leave it up to you as it's a small change anyway.
